src/test/index.ts

```diff
@@ -31,8 +31,86 @@ type TestFixtures = PlaywrightTestArgs & PlaywrightTestOptions & {
   _browserType: BrowserType;
   _artifactsDir: () => string,
+  _reuseBrowerContext: ReuseBrowerContextStorage,
 };
 
+class ReuseBrowerContextStorage {
+  private _browserContext?: BrowserContext;
+  private _uniqueOrigins = new Set<string>();
+
+  isEnabled(): boolean {
+    return !!process.env.PWTEST_REUSE_CONTEXT;
+  }
+
+  hasContext(): boolean {
+    return !!this._browserContext;
+  }
+
+  setContext(context: BrowserContext) {
+    assert(!this._browserContext);
+    this._browserContext = context;
+    this._browserContext.on('page', page => page.on('framenavigated', frame => this._uniqueOrigins.add(new URL(frame.url()).origin)));
+  }
+
+  async obtainContext(newContextOptions: BrowserContextOptions): Promise<BrowserContext> {
+    assert(this._browserContext);
+    let pages = this._browserContext.pages();
+    const page = pages[0];
+    if (this._uniqueOrigins.size > 0) {
+      let pageWorker = null;
+      if (pages.length > 1) {
+        await page.bringToFront();
+        pageWorker = pages[1];
+      } else {
+        pageWorker = await this._browserContext.newPage();
+        pages = this._browserContext.pages();
+      }
+      for (const origin of this._uniqueOrigins) {
+        await pageWorker.route(origin, route => route.fulfill({body: `<html></html>`, contentType: 'text/html'}), { times: 1 });
+        await pageWorker.goto(origin);
+        await pageWorker.evaluate(() => window.localStorage.clear());
+        await pageWorker.evaluate(() => window.sessionStorage.clear());
+      }
+    }
+    for (const p of pages.slice(1))
+      await p.close();
+    await this._browserContext.clearCookies();
+    await this._applyNewContextOptions(page, newContextOptions);
+    await page.goto('about:blank');
+    this._uniqueOrigins.clear();
+    return this._browserContext;
+  }
+
+  private async _applyNewContextOptions(page: Page, newOptions: BrowserContextOptions) {
+    const oldOptions: BrowserContextOptions = (this._browserContext as any)._options;
+    if (
+      (
+        oldOptions.viewport?.width !== newOptions.viewport?.width ||
+        oldOptions.viewport?.height !== newOptions.viewport?.height
+      ) &&
+      (newOptions.viewport?.height && newOptions.viewport?.width)
+    )
+      await page.setViewportSize({width: newOptions.viewport?.width, height: newOptions.viewport?.height });
+    const emulateMediaOptions: Partial<Parameters<Page['emulateMedia']>[0]> = {};
+    if (oldOptions.colorScheme !== newOptions.colorScheme)
+      emulateMediaOptions.colorScheme = newOptions.colorScheme;
+    if (oldOptions.forcedColors !== newOptions.forcedColors)
+      emulateMediaOptions.forcedColors = newOptions.forcedColors;
+    if (oldOptions.reducedMotion !== newOptions.reducedMotion)
+      emulateMediaOptions.reducedMotion = newOptions.reducedMotion;
+    if (Object.keys(emulateMediaOptions).length > 0)
+      await page.emulateMedia(emulateMediaOptions);
+    (this._browserContext as any)._options = newOptions;
```

Comment:
This does not really do nothing, only makes Playwright break inside. You should remember settings for your context and only reuse the context if the settings match.